domaincommon/src/state.rs

```diff
@@ -562,14 +638,76 @@ impl Match {
     ) -> Result<Vec<<Match as Aggregate>::Event>> {
         let pings =
             crate::radar::radar_ping(state, &state.mechs[mech].position, &state.mechs[mech].team);
-        Ok(vec![GameEvent::RadarScanCompleted {
-            actor: mech.to_string(),
-            results: pings,
-        },
-        GameEvent::ActionPointsConsumed {
-            mech: mech.to_string(),
-            points_consumed: cmd.action_points(),
-        },])
+        Ok(vec![
+            GameEvent::RadarScanCompleted {
+                actor: mech.to_string(),
+                results: pings,
+            },
+            GameEvent::ActionPointsConsumed {
+                mech: mech.to_string(),
+                points_consumed: cmd.action_points(),
+            },
+        ])
+    }
+
+    fn handle_register_update(
+        state: &<Match as Aggregate>::State,
+        cmd: &MechCommand,
+    ) -> Result<Vec<<Match as Aggregate>::Event>> {
+        let event = if let MechCommand::RegisterUpdate { mech, reg, op, .. } = cmd {
+            let curr_val = state.mechs[mech].registers.get(reg);
+            match op {
+                RegisterOperation::Accumulate(acc) if *reg == EAX || *reg == ECX => {
+                    if let Some(RegisterValue::Number(n)) = curr_val {
+                        // Prevent positive overflow
```

Comment:
Check out the `saturating` operations on the `u64` type. They saturate at numeric bounds so they'll automatically stop at the edges instead of overflowing. Might allow you to get rid of at least 1 pesky if statement

https://doc.rust-lang.org/std/primitive.u64.html#method.saturating_add

Reply:
Awesome, those if statements did feel weird to write out.